STX.REST.RESTFulSense.Clients.Tests.Unit/Services/Foundations/HttpExchanges/HttpExchangeServiceTests.Headers.cs

```diff
@@ -0,0 +1,1778 @@
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            string[] invalidTrailerHeaders = CreateInvalidStringArrayHeaders();
+            invalidTrailerHeaders
+                .Select(invalidTrailerHeader =>
+                {
+                    theoryData.Add(
+                        CreateTrailerHeaderException(
+                            new string[] { invalidTrailerHeader }));
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            TransferCodingHeader[] invalidTransferEncondingHeaders = CreateInvalidTransferCodingHeaders();
+            invalidTransferEncondingHeaders
+                .Select(invalidTransferEncondingHeader =>
+                {
+                    theoryData.Add(
+                        CreateTransferEncodingException(
+                            new TransferCodingHeader[] { invalidTransferEncondingHeader }));
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            ProductHeader[] invalidUpgradeHeaders = CreateInvalidProductHeaders();
+            invalidUpgradeHeaders
+                .Select(invalidUpgradeHeader =>
+                {
+                    theoryData.Add(
+                        CreateUpgradeHeaderException(
+                            new ProductHeader[] { invalidUpgradeHeader }));
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            ProductInfoHeader[] invalidUserAgentHeaders = CreateInvalidProductInfoHeaders();
+            invalidUserAgentHeaders
+                .Select(invalidUserAgentHeader =>
+                {
+                    theoryData.Add(
+                        CreateUserAgentHeaderException(
+                            new ProductInfoHeader[] { invalidUserAgentHeader }));
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            ViaHeader[] invalidViaHeaders = CreateInvalidViaHeaders();
+            invalidViaHeaders
+                .Select(invalidViaHeader =>
+                {
+                    theoryData.Add(
+                        CreateViaHeaderException(
+                            new ViaHeader[] { invalidViaHeader }));
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            WarningHeader[] invalidWarningHeaders = CreateInvalidWarningHeaders();
+            invalidWarningHeaders
+                .Select(invalidWarningHeader =>
+                {
+                    theoryData.Add(
+                        CreateWarningHeaderException(
+                            new WarningHeader[] { invalidWarningHeader }));
+
+                    return theoryData;
+                })
+                .ToArray();
+
+            return theoryData;
+        }
+
+        private static TheoryData<dynamic> CreateRequestContentHeadersValidationExceptions(TheoryData<dynamic> theoryData)
```

Comment:
ContentLength, ContentLocation, ContentMD5, Expires and LastModified we need to ensure if we could add some validation

Reply:
About this, do you mean going further to check the documentation or native implementation if it needs validation?